code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/BasicTemplateNotificationBuilder.kt

```diff
@@ -0,0 +1,340 @@
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.core.R
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ServiceProvider
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a basic push template notification.
+ */
+internal object BasicTemplateNotificationBuilder : AEPPushTemplateNotificationBuilder() {
+    private const val SELF_TAG = "BasicTemplateNotificationBuilder"
+    private const val ACTION_BUTTON_CAPACITY = 3
+
+    @Throws(NotificationConstructionFailedException::class)
+    fun construct(
+        context: Context,
+        pushTemplate: BasicPushTemplate?,
+        trackerActivityName: String?,
+        broadcastReceiverName: String?
+    ): NotificationCompat.Builder {
+        if (pushTemplate == null) {
+            throw NotificationConstructionFailedException(
+                "push template is null, cannot build a basic template notification."
+            )
+        }
+        val cacheService = ServiceProvider.getInstance().cacheService
+            ?: throw NotificationConstructionFailedException("Cache service is null, basic template notification will not be constructed.")
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building a basic template push notification."
+        )
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_expanded)
+
+        // get push payload data
+        val imageUri = pushTemplate.imageUrl
+        val pushImage = downloadImage(cacheService, imageUri)
+        pushImage?.let {
+            expandedLayout.setImageViewBitmap(R.id.expanded_template_image, pushImage)
+        }
+        smallLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        smallLayout.setTextViewText(R.id.notification_body, pushTemplate.body)
+        expandedLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        expandedLayout.setTextViewText(
+            R.id.notification_body_expanded, pushTemplate.expandedBodyText
+        )
+
+        // Create the notification channel if needed
+        channelIdToUse = createChannelAndGetChannelID(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance()
+        )
+
+        // create the notification builder with the common settings applied
+        val builder =
+            super.construct(context, pushTemplate, trackerActivityName, smallLayout, expandedLayout)
+
+        // add any action buttons defined for the notification
+        addActionButtons(
+            context,
+            trackerActivityName,
+            builder,
```

Comment:
Instead of passing around the `builder` object, how do you feel about making this an extension function for `NotificationCompat.Builder` inside `AEPPushTemplateNotificationBuilder` ? Also avoids passing around the builder in `LegacyNotificationBuilder`
 Thoughts @prudrabhat?